### PR TITLE
Use LogFields interface for Adding Events

### PR DESCRIPTION
**Pull request by pquerna**

### CODE REVIEW

File: nethttp/client.go [hunk 1]
```diff
@@ -197,54 +198,79 @@ func (h *Tracer) clientTrace() *httptrace.ClientTrace {
...
 
 func (h *Tracer) wait100Continue() {
-	h.sp.LogEvent("Wait 100 continue")
+	h.sp.LogFields(log.String("event", "Wait100Continue"))
 }
 
 func (h *Tracer) wroteRequest(info httptrace.WroteRequestInfo) {
 	if info.Err != nil {
+		h.sp.LogFields(
+			log.String("event", "WroteRequest"),
```
Comment (yurishkuro) [inline nethttp/client.go]: I think event should be `error` here - see https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table

Comment (pquerna) [inline nethttp/client.go]: hmm, I think `log.Error(info.Err)` does exactly that?

**Comment by yurishkuro** [comment]
@dominikh any objections?

**Comment by dominikh** [comment]
LGTM. Want to merge?

**Pull request merged**